pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/KitRequestShipping.java

```diff
@@ -1661,54 +1669,131 @@ public static List<KitRequestShipping> getKitRequestsAfterBookmark(long bookmark
+        if (isGen2MigratedKit(ddpInstance, ddpParticipantId, baseUrl)) {
+            // if kit uploaded with a gen2 legacy id, then use that id to generate the collaborator id
+            return handleGen2MigratedKit(instanceId, ddpParticipantId, shortId, collaboratorIdPrefix,
+                    collaboratorParticipantLengthOverwrite);
+        }
+        return generateCollaboratorParticipantId(shortId, ddpParticipantId, collaboratorIdPrefix, collaboratorParticipantLengthOverwrite);
+    }
+
+    /**
+     * Checks if the participant has a legacy shorty id and if so, creates a collaborator participant id based on that.
+     * If the participant has previous kits using that id, then returns that id. If the participant is not a legacy participant
+     * or does not have kits with legacy id, then returns null.
+     *
+     * @param ddpInstance the DDPInstance of the participant
+     * @param shortId the Pepper shortId
+     * @param collaboratorParticipantLengthOverwrite the length of the collaborator participant id
+     * */
+    private static String getLegacyCollaboratorParticipantId(DDPInstance ddpInstance, String shortId,
+                                                             String collaboratorParticipantLengthOverwrite) {
+        String legacyCollaboratorPtId = findLegacyCollaboratorParticipantId(ddpInstance, shortId,
+                collaboratorParticipantLengthOverwrite);
+        if (StringUtils.isNotBlank(legacyCollaboratorPtId)
+                && !kitRequestDao.getKitRequestsForCollaboratorParticipantId(legacyCollaboratorPtId).isEmpty()) {
+            return legacyCollaboratorPtId;
+        }
+        return null;
+    }
+
+    /**
+     * Checks if the legacyCollaboratorParticipantId is not null or empty
+     * */
+    private static boolean legacyParticipantIdExists(String legacyCollaboratorParticipantId) {
+        return StringUtils.isNotBlank(legacyCollaboratorParticipantId);
+    }
+
+    private static String handleGen2MigratedKit(int instanceId, String ddpParticipantId, String shortId, String collaboratorIdPrefix,
+                                                String overwrite) {
+        String collaboratorId = KitUtil.getKitCollaboratorId(ddpParticipantId, instanceId);
+        if (StringUtils.isNotBlank(collaboratorId)) {
+            return collaboratorId;
+        }
+        // if a gen2 migrated kit without previous collab id, then use the sample id to extract the participant id
+        String sampleId = KitUtil.getTissueCollaboratorId(ddpParticipantId, instanceId);
+        if (StringUtils.isBlank(sampleId)) {
+            return handleBlankSampleId(shortId, collaboratorIdPrefix, overwrite);
+        }
+
+        return extractParticipantIdFromSampleId(sampleId);
+    }
+
+    private static String handleBlankSampleId(String shortId, String collaboratorIdPrefix, String overwrite) {
```

Comment:
This method just got the same parts of the old code encapsulated as a method for the ease of readability